### ISSUE TITLE
RxApollo return error in retryWhen operator

### ISSUE THREAD

**Issue by svyatoslav-reshetnikov** [apollographql/apollo-android#963]
I have a problem with ```retryWhen``` operator with ```RxApollo```. I want to call query and repeat it again. I called ```retryWhen``` operator in my Rx sequence and I got an error "Call is cancelled".

Sample of code (Kotlin):
```
rxApolloCall()
.flatMap { apolloCall ->
    if (apolloCall.hasErrors()) {
        Flowable.error<Exception>(Exception())
    } else {
        Flowable.just(apolloCall)
    }
}
.retryWhen { sequence ->
    sequence
            .flatMap { exception ->
                if (exception is Exception) {
                    Flowable.timer(1, TimeUnit.SECONDS)
                } else {
                    Flowable.error(exception)
                }
            }
   }
}
.subscribe(
    { result ->  Log,e(result) },
    { error -> Log.e("TAG", "{$error.message}")},
    { Log.d("TAG", "completed") }
)
```

I'm using 0.5.0 version of apollo android support, apollo runtime and apollo rx 2.

**Comment by svyatoslav-reshetnikov** [comment]
I solved my problem with other GraphQL library. Thanks.

**Comment by SeanBeckman** [comment]
Sorry to resurrect this question, but how exactly do you achieve this then? The following simplified example is falling into the AlreadyExecuted IllegalStateException:
`val call = apolloClient
            .query(query)
        return Rx2Apollo.from(call.clone())
            .retry(2)`

Where are we actually supposed to clone the call if Rx2Apollo is enqueuing for us and Rx's retry operator just resubscribes to the stream based on a call which has fallen into an active state.

**Comment by sh3lan93** [comment]
I have used the solution of cloning the call using Rx3Aollo but same error occurred **Already executed**

**Comment by martinbonnin** [comment]
Indeed the cloning would certainly have to happen inside the `subscribe()` like we did for [coroutines](https://github.com/apollographql/apollo-android/commit/fca0b5c825f16eeccd392de6f21575c24be51c3e).

**Comment by sh3lan93** [comment]
Here is how I clone
```
apolloClient.mutate(CreatEmptyCartMutation()).clone().let {
   Single.fromObservable(Rx3Apollo.from(it)).addHandlingTooManyRequests().map { it.data?.createEmptyCart }
}
```
then I subscribed what should I do to overcome the **Already executed**

**Comment by martinbonnin** [comment]
Yup, sorry your code looks good. I meant the clone should happen in the Apollo code (most likely [here](https://github.com/apollographql/apollo-android/blob/3d7e5ffcc1cfa7c0428f5bb42084e8919fbacea9/apollo-rx2-support/src/main/java/com/apollographql/apollo/rx2/Rx2Apollo.java#L93)). I'll make a fix.

**Comment by martinbonnin** [comment]
@sh3lan93 The fix is in the 2.4.1-SNAPSHOT (See https://github.com/apollographql/apollo-android#snapshots for how to include snapshots repositories). Any chance you can try it and confirm if it works?

**Comment by sh3lan93** [comment]
@martinbonnin yes it working fine now 👏
